### DISCUSSION TITLE
Getting started - installing Nemo on Windows

### DISCUSSION THREAD

**Discussion by jnmaloney** [NVIDIA-NeMo/NeMo#6515] [General]
I'm following the instructions to install Nemo on Windows. So far I've set up a fresh Conda environment and installed the prereqs. I'm failing on `python -m pip install git+https://github.com/NVIDIA/NeMo.git@r1.17.0#egg=nemo_toolkit[all]`

I'm getting the error 
```
      cl : Command line error D8021 : invalid numeric argument '/Wno-register'
      error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Professional\\VC\\Tools\\MSVC\\14.29.30133\\bin\\HostX86\\x64\\cl.exe' failed with exit code 2
```
so it's picking up my MSVC toolchain, which might not be intentional? I'm not sure what this step is supposed to do.

**Comment by titu1994** [comment]
We don't support windows officially via pip. We do have a few users who were able to patch some installation steps to do it, but for a better experience I'd suggest using WSL 2 on windows

**Comment by windowshopr** [comment]
Download the repo/zip file locally, unzip it somewhere, open a terminal/powershell window, `cd` to that folder, and run `python setup.py install`. Worked for me on Windows 10, Python 3.8. Pip install isn't supported for Windows yet as @titu1994 mentioned, but building from source like this is fine.

**Reply by trzy** [reply]
```
Installed c:\users\bart\anaconda3\envs\nemo\lib\site-packages\nemo_toolkit-1.21.0rc0-py3.11.egg
Processing dependencies for nemo-toolkit==1.21.0rc0
Searching for triton
Reading https://pypi.org/simple/triton/
No local packages or working download links found for triton
error: Could not find suitable distribution for Requirement.parse('triton')
```

That doesn't work here.

**Reply by windowshopr** [reply]
I'm not using anaconda in my env so might be different for conda users. If you figure it out, post your steps!
